app/Models/Search.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/**
+ * Contains a search from the search form.
+ *
+ * It allows to extract meaningful bits of the search and store them in a
+ * convenient object
+ */
+class FreshRSS_Search {
+
+	// This contains the user input string
+	private $raw_input;
+	// The following properties are extracted from the raw input
+	private $intitle;
+	private $min_date;
+	private $max_date;
+	private $min_pubdate;
+	private $max_pubdate;
+	private $inurl;
+	private $author;
+	private $tags;
+	private $search;
+
+	public function __construct($input) {
+		if (strcmp($input, '') == 0) {
+			return;
+		}
+		$this->raw_input = $input;
+		$input = $this->parseIntitleSearch($input);
+		$input = $this->parseAuthorSearch($input);
+		$input = $this->parseInurlSearch($input);
+		$input = $this->parsePubdateSearch($input);
+		$input = $this->parseDateSearch($input);
+		$input = $this->parseTagsSeach($input);
+		$this->search = $input;
+	}
+
+	public function getRawInput() {
+		return $this->raw_input;
+	}
+
+	public function getIntitle() {
+		return $this->intitle;
+	}
+
+	public function getMinDate() {
+		return $this->min_date;
+	}
+
+	public function getMaxDate() {
+		return $this->max_date;
+	}
+
+	public function getMinPubdate() {
+		return $this->min_pubdate;
+	}
+
+	public function getMaxPubdate() {
+		return $this->max_pubdate;
+	}
+
+	public function getInurl() {
+		return $this->inurl;
+	}
+
+	public function getAuthor() {
+		return $this->author;
+	}
+
+	public function getTags() {
+		return $this->tags;
+	}
+
+	public function getSearch() {
+		return $this->search;
+	}
+
+	/**
+	 * Parse the search string to find intitle keyword and the search related
+	 * to it.
+	 * The search is the first word following the keyword.
+	 *
+	 * @param string $input
+	 * @return string
+	 */
+	private function parseIntitleSearch($input) {
+		if (preg_match('/intitle:(?P<delim>[\'"])(?P<search>.*)(?P=delim)/U', $input, $matches)) {
+			$this->intitle = $matches['search'];
+			$input = str_replace($matches[0], '', $input);
+		} else if (preg_match('/intitle:(?P<search>\w*)/', $input, $matches)) {
+			$this->intitle = $matches['search'];
+			$input = str_replace($matches[0], '', $input);
+		}
+		return $this->cleanSearch($input);
```

Comment:
Is it really useful to call `cleanSearch()` each time? We may call it once in constructor before `$this->search = $input;`, don't you think?